### PR TITLE
WIP: circleci basic build configuration for macOS, fixes #575

### LINKED ISSUE #575: Use Circle CI for macOS testing
## What happened (or feature request):

CircleCI now supports building in multiple environments from the same project (https://discuss.circleci.com/t/how-to-use-circleci-for-linux-xcode-in-one-project/693/6). We should consider migrating macOS testing to Circle and deprecate mac surf testing.

### PR DESCRIPTION
**Pull request by rfay**
## The Problem/Issue/Bug:

We've been using surf-build to build on darwin, and we wish we had something better. Circleci supports darwin, but it requires some workarounds for docker.

## How this PR Solves The Problem:

This uses some tidbits found in various places to actually install and start docker. It might even work.

The current status is way less than done, it's just crude attempts to demonstrate it working. In reality we'll have to integrate into current Circleci workflow, and make this part cleaner.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,179 +1,31 @@
...
-      # Now build using the regular ddev-only technique - this results in a fully clean set of executables.
-      - run:
-          command: make -s clean linux darwin windows
-          name: Build the ddev executables
-
-      - run:
-          command: bin/linux/ddev version
-          name: ddev version information
+      - run: |
+          brew cask install docker && brew install docker docker-compose golang
```
Comment (cweagans) [inline .circleci/config.yml]: You shouldn't have to `brew install docker docker-compose` here - the docker client and docker-compose both ship with DfM.

Comment (rfay) [reply .circleci/config.yml]: Agreed. However, the install technique I'm doing there doesn't seem to get that in there. Or maybe it was one of those things where I was just confused.

**Comment by rfay** [comment]
The basic problem with this right now is that the 
```
          command: /Applications/Docker.app/Contents/MacOS/Docker --unattended >/tmp/docker.log 2>&1
           background: true
```
when run with sudo creates a docker environment where mounts are owned by root (on the host fs). And if you run it without sudo it doesn't succeed. I'm not sure if there's some magic incantation of /Applications/Docker.app/Contents/MacOS/Docker that would specify the user. I haven't been able to find any documentation of it. Don't think it's open-source at all.

**Comment by rfay** [comment]
Will let this simmer and reopen when we know more.

**Pull request closed without merging**